compiler/rustc_expand/src/proc_macro_server.rs

```diff
@@ -421,13 +427,58 @@ impl server::TokenStream for Rustc<'_> {
         parse_stream_from_source_str(
             FileName::proc_macro_source_code(src),
             src.to_string(),
-            self.sess,
+            self.sess(),
             Some(self.call_site),
         )
     }
     fn to_string(&mut self, stream: &Self::TokenStream) -> String {
         pprust::tts_to_string(stream)
     }
+    fn expand_literal(&mut self, stream: &Self::TokenStream) -> Result<Self::Literal, ()> {
+        // Parse the expression from our tokenstream.
+        let expr = {
+            let mut p = rustc_parse::stream_to_parser(
+                self.sess(),
+                stream.clone(),
+                Some("proc_macro expand literal"),
+            );
+            let expr = p.parse_expr().map_err(|_| ())?;
```

Comment:
```suggestion
            let expr = p.parse_expr().map_err(|err| { err.cancel(); })?;
```
This is going to panic on drop if the error is not reported or canceled.
Not sure what is the preferred behavior, should `expand_literal` be able to accept garbage and return `Err`, with the user ignoring `Err` and continuing compilation?

Reply:
I'm unsure about that. I considered making every `Err` returned also correspond with an error diagnostic being emitted, so the build will fail after that point. My reasoning was that this would communicate the most useful diagnostics to the user without exposing internal details to the calling macro. I didn't end up following through with the idea as it wasn't consistent with the behaviour of other APIs in the proc_macro interface, and I worried about emitting duplicate errors as the macro will likely panic itself. I'm open to making the change though.